imports/plugins/core/orders/server/no-meteor/resolvers/Order/orderSummary.js

```diff
@@ -0,0 +1,80 @@
+import { xformRateToRateObject } from "@reactioncommerce/reaction-graphql-xforms/core";
+
+/**
+ * @name "Order.orderSummary"
+ * @method
+ * @memberof Order/GraphQL
+ * @summary Returns an aggregate of all fulfillmentGroup summaries to provide a single orderSummary
+ * @param {Object} context An object with request-specific state
+ * @param {Object} order - Result of the parent resolver, which is a Order object in GraphQL schema format
+ * @return {Object} An object containing order pricing information from all fulfillmentGroups
+ */
+export default async function orderSummary(context, order) {
+  const { shipping: fulfillmentMethods } = order;
+  const totalDiscounts = [];
+  const totalShipping = [];
+  const totalSubtotal = [];
+  const totalSurcharges = [];
+  const totalTaxableAmount = [];
+  const totalTaxes = [];
+  const totalTotal = [];
+
+  // Loop over each fulfillmentGroup (shipping[]), and push all values into `totalX` array
+  fulfillmentMethods.forEach((fulfillmentMethod) => {
+    const { invoice: { discounts, shipping, subtotal, surcharges, taxableAmount, taxes, total } } = fulfillmentMethod;
+
+    totalDiscounts.push(discounts);
+    totalShipping.push(shipping);
+    totalSubtotal.push(subtotal);
+    totalSurcharges.push(surcharges);
+    totalTaxableAmount.push(taxableAmount);
+    totalTaxes.push(taxes);
+    totalTotal.push(total);
+  });
+
+  // Reduce each `totalX` array to get order total from all fulfillmentGroups
+  const totalDiscountsAmount = totalDiscounts.reduce((acc, value) => acc + value, 0);
+  const totalShippingAmount = totalShipping.reduce((acc, value) => acc + value, 0);
+  const totalSubtotalAmount = totalSubtotal.reduce((acc, value) => acc + value, 0);
+  const totalSurchargesAmount = totalSurcharges.reduce((acc, value) => acc + value, 0);
+  const totalTaxableAmountAmount = totalTaxableAmount.reduce((acc, value) => acc + value, 0);
+  const totalTaxesAmount = totalTaxes.reduce((acc, value) => acc + value, 0);
+  const totalTotalAmount = totalTotal.reduce((acc, value) => acc + value, 0);
+
+  // Get currencyCode from first fulfilllmentMethod
+  const { currencyCode } = fulfillmentMethods[0];
```

Comment:
@aldeed Not sure if this is the best way to get `currencyCode`. Was thinking we could also query and get the entire shop, and then take it from `shop.currency.code`, but didn't really want to add another query for the `Shops` collection when this data is available here.

My only qualm with it is what if we have more than one fulfillmentMethod. They both should probably be the same currency code, right?

Reply:
I don't think it's possible to have groups with different currencies, but to avoid the `[0]`, you can use `order.currencyCode`, which is required so should always be set.